ext/src/ruby_api/func.rs

```diff
@@ -127,25 +132,25 @@ impl<'a> Func<'a> {
     /// @yard
     /// @return [Array<Symbol>] The function's parameter types.
     pub fn params(&self) -> Result<RArray, Error> {
-        let params = self
-            .inner
-            .ty(self.store.context()?)
-            .params()
-            .map(ToSym::to_sym)
-            .collect();
-        Ok(params)
+        let ty = self.inner.ty(self.store.context()?);
+        let len = ty.params().len();
+        let mut params = ty.params();
+        params.try_fold(RArray::with_capacity(len), |array, p| {
+            array.push(p.as_sym()?).map_err(|e| error!("{e}"))?;
+            Ok::<_, Error>(array)
+        })
     }
 
     /// @yard
     /// @return [Array<Symbol>] The function's result types.
     pub fn results(&self) -> Result<RArray, Error> {
-        let results = self
-            .inner
-            .ty(self.store.context()?)
-            .results()
-            .map(ToSym::to_sym)
-            .collect();
-        Ok(results)
+        let ty = self.inner.ty(self.store.context()?);
+        let len = ty.results().len();
+        let mut results = ty.results();
+        results.try_fold(RArray::with_capacity(len), |array, r| {
+            array.push(r.as_sym()?).map_err(|e| error!("{e}"))?;
+            Ok::<_, Error>(array)
```

Comment:
```suggestion
            array.push(r.as_sym()?)?;
            Ok(array)
```


1. if `Array#push` fails (it should never because it's not frozen), we wouldn't want to raise a `Wasmtime::Error` but let the original error bubble, so I don't think we want the `map_err`.
2. Looks like the compiler figured out the return type just fine with `Ok(array)` on my side.

Same for params.

Reply:
Fixed!